api/v1alpha1/databasecluster_types.go

```diff
@@ -326,6 +330,8 @@ type Sharding struct {
 	Shards int32 `json:"shards"`
 	// ConfigServer represents the sharding configuration server settings
 	ConfigServer ConfigServer `json:"configServer"`
+	// Affinity defines scheduling affinity.
+	Affinity *corev1.Affinity `json:"affinity,omitempty"`
```

Comment:
This should be moved inside the ConfigServer struct as it only applies to that component instead of the whole sharding config.

Reply:
updated